ports/atmel-samd/common-hal/displayio/FourWire.c

```diff
@@ -0,0 +1,147 @@
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Scott Shawcroft for Adafruit Industries
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "shared-bindings/displayio/FourWire.h"
+
+#include <stdint.h>
+
+#include "shared-bindings/busio/SPI.h"
+#include "shared-bindings/digitalio/DigitalInOut.h"
+
+#include "tick.h"
+
+void common_hal_displayio_fourwire_construct(displayio_fourwire_obj_t* self,
+    const mcu_pin_obj_t* clock, const mcu_pin_obj_t* data, const mcu_pin_obj_t* command,
+    const mcu_pin_obj_t* chip_select, const mcu_pin_obj_t* reset, uint16_t width,
+    uint16_t height, int16_t colstart, int16_t rowstart, uint16_t color_depth,
+    uint8_t set_column_command, uint8_t set_row_command, uint8_t write_ram_command) {
+
+    common_hal_busio_spi_construct(&self->bus, clock, data, mp_const_none);
+    common_hal_digitalio_digitalinout_construct(&self->command, command);
+    common_hal_digitalio_digitalinout_switch_to_output(&self->command, true, DRIVE_MODE_PUSH_PULL);
+    common_hal_digitalio_digitalinout_construct(&self->chip_select, chip_select);
+    common_hal_digitalio_digitalinout_switch_to_output(&self->chip_select, true, DRIVE_MODE_PUSH_PULL);
+
+    common_hal_digitalio_digitalinout_construct(&self->reset, reset);
+    common_hal_digitalio_digitalinout_switch_to_output(&self->reset, true, DRIVE_MODE_PUSH_PULL);
+
+    self->width = width;
+    self->height = height;
+    self->color_depth = color_depth;
+    self->set_column_command = set_column_command;
+    self->set_row_command = set_row_command;
+    self->write_ram_command = write_ram_command;
+    self->current_group = NULL;
+    self->colstart = colstart;
+    self->rowstart = rowstart;
+}
+
+bool common_hal_displayio_fourwire_begin_transaction(displayio_fourwire_obj_t* self) {
+    if (!common_hal_busio_spi_try_lock(&self->bus)) {
+        return false;
+    }
+    common_hal_busio_spi_configure(&self->bus, 12000000, 0, 0, 8);
```

Comment:
I'd vote against hardcoding the SPI frequency, because for example nRF can do 32MHz

Reply:
Totally. This was just there to get it going. I need to work on how this is shared with other ports as well.

I added a TODO because this PR is aimed at establishing the group hierarchy stuff and not perfecting the display API and its internals.